Sources/xcodeproj/Objects/Project/PBXProject.swift

```diff
@@ -125,15 +125,15 @@ public final class PBXProject: PBXObject {
     }
 
     /// Package references.
-    var packageReferences: [PBXObjectReference]?
```

Comment:
What we've been doing for deciding whether a property should be optional or not is whether Xcode is able to open a project without that attribute. If I'm not wrong, Xcode is able to open a project that doesn't have this attribute. If that's the case, I'd keep this property as optional. It also ensures that we are not adding attributes to Xcode 10 projects that are are not readable by Xcode 10.

Reply:
I just changed it to an empty array if there is nothing to decode, and only encode if there's something in it. I'm not sure I understand you, does this mess up a specific use case?

Reply:
I'd say this change makes sense - optional arrays are often troubling when it comes to accessing their contents. I can't see a situation where this will cause a problem.

If you want to check the presence of a package inside the Xcodeproj it's possible to by calling `packageReferences.count > 0`.

Reply:
Let's take the following scenario:
- The user edits a Xcode project that doesn't have the `packageReferences` attribute.
- We decode the attribute to an empty array.
- After saving the project, the user will see in the diff that a new attribute is added.

Xcode 10.x will ignore the attribute, but we are introducing unexpected modifications to the project. That's something we've always avoided, and I don't think it's solvable by not writing the array when it's empty, because that's another possible value for the attribute.

If I'm not wrong, I think Xcode 11 updates some attribute in the project to say that the project is compatible with version X of Xcode. What about using that value to decide whether the attribute should be written or not?

Reply:
@pepibumur What do you mean by attribute? The packageReferences? If the array is empty we don't write it to the project, exactly as if it was nil

Reply:
Yes, let me put it on a table:

| Project in disk             | Read project                | Diff? |
|-----------------------------|-----------------------------|-------|
|  No packageReferences       | Empty packageReferences     | Yes   |
| Empty packageReferences     | Empty packageReferences     | No    |
| Non-empty packageReferences | Non-empty packageReferences | No    |

By turning nil into an empty array we'll get a diff after writing the project to disk in the first scenario.

Reply:
I assume'd we wouldn't write `Empty packageReferences`. Is there a benefit to doing so? If we assume empty means we don't write it then we don't have to worry about the truth table above since we only handle two scenarios, empty and non-empty

Reply:
There isn't any difference for Xcode. The point is that users don't want to see git diffs if the project hasn't been modified, and with the current implementation they will. I suggest the following modification:

```swift
// This one remains optional
var packageReferences: [PBXObjectReference]?

// We define this one as non-optional
public var packages: [XCRemoteSwiftPackageReference] {
  set {
    packageReferences = newValue.references()
  }
  get {
     return packageReferences.objects() ?? []
  }
}
```

Does that look good @ollieatkinson / @yonaskolb ?

Reply:
That makes sense - 👍